index.js

```diff
@@ -48,13 +49,72 @@ module.exports = {
           ), self.findOrCreateUser(spec));
           spec.name = dummy.name;
         }
+        spec.label = spec.label || spec.name;
         spec.options.callbackURL = self.getCallbackUrl(spec, true);
         self.strategies[spec.name] = new Strategy(spec.options, self.findOrCreateUser(spec));
         self.apos.login.passport.use(self.strategies[spec.name]);
         self.addLoginRoute(spec);
         self.addCallbackRoute(spec);
         self.addFailureRoute(spec);
       });
+
+      self.on('apostrophe-login:after', 'redirectToNewLocale', async req => {
+        const workflow = self.apos.modules['apostrophe-workflow'];
+        if (!workflow) {
+          return;
+        }
+        const cacheSet = util.promisify(workflow.crossDomainSessionCache.set);
+        const deserializeUser = util.promisify(self.apos.login.deserializeUser);
+        if (!req.session.passportWorkflow) {
+          return;
+        }
+        const {
+          oldLocale,
+          newLocale,
+          oldSlug
+        } = req.session.passportWorkflow;
+        delete req.session.passportWorkflow;
+        const crossDomainSessionToken = self.apos.utils.generateId();
+        await cacheSet(crossDomainSessionToken, JSON.stringify(req.session), 60 * 60);
+        req.user = await deserializeUser(req.user._id);
+        let doc = await self.apos.docs.find(req, {
```

Comment:
Some grinding through tedium here because link-to-locale works in terms of slugs, and needs to know the slug in the new locale, which could be different.